src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/UnstructuredAddress.java

```diff
@@ -40,12 +43,24 @@ public static UnstructuredAddress fromString(final String edifact) {
             System.arraycopy(splitByColon, 1, addressLines, 0, splitByColon.length - 1);
         }
 
-        final String postCode = splitByPlus[INDEX_POSTCODE];
+        String postCode = null;
+        if (splitByPlus.length >= INDEX_POSTCODE && StringUtils.isNotBlank(splitByPlus[INDEX_POSTCODE])) {
+            postCode = splitByPlus[INDEX_POSTCODE];
+        }
+
         final String qualifier = edifact.startsWith(KEY + PLUS_SEPARATOR + PLUS_SEPARATOR + FORMAT) ? FORMAT : "";
 
         return new UnstructuredAddress(qualifier, addressLines, postCode);
     }
 
+    public Optional<String[]> getAddressLines() {
+        return Optional.ofNullable(addressLines);
+    }
```

Comment:
Just as we wouldn't normally return `Optional<Collection<T>>`, preferring to return an empty collection instead, I suggest returning the empty array here too.

```suggestion
    public String[] getAddressLines() {
        return addressLines != null ? addressLines : new String[0];
    }
```

Reply:
Sure, fixed in e53eeadfa2c7af1ae5a59c4d497f89586dca11cf